File: SDKs/zuora.com/2021-08-20/python/src/sdk/models/operations/action_postcreate.py
import dataclasses
from typing import Optional
from ..shared import proxyactioncreaterequest as shared_proxyactioncreaterequest
from ..shared import proxyunauthorizedresponse as shared_proxyunauthorizedresponse
from ..shared import saveresult as shared_saveresult


@dataclasses.dataclass
class ActionPosTcreateQueryParams:
    reject_unknown_fields: Optional[bool] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'rejectUnknownFields', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class ActionPosTcreateHeaders:
    x_zuora_wsdl_version: Optional[str] = dataclasses.field(default=None, metadata={'header': { 'field_name': 'X-Zuora-WSDL-Version', 'style': 'simple', 'explode': False }})
    zuora_entity_ids: Optional[str] = dataclasses.field(default=None, metadata={'header': { 'field_name': 'Zuora-Entity-Ids', 'style': 'simple', 'explode': False }})
    zuora_track_id: Optional[str] = dataclasses.field(default=None, metadata={'header': { 'field_name': 'Zuora-Track-Id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class ActionPosTcreateRequest:
    headers: ActionPosTcreateHeaders = dataclasses.field()
    query_params: ActionPosTcreateQueryParams = dataclasses.field()
    request: shared_proxyactioncreaterequest.ProxyActioncreateRequest = dataclasses.field(metadata={'request': { 'media_type': 'application/json' }})
    

@dataclasses.dataclass
class ActionPosTcreateResponse:
    content_type: str = dataclasses.field()
    headers: dict[str, list[str]] = dataclasses.field()
    status_code: int = dataclasses.field()
    proxy_unauthorized_response: Optional[shared_proxyunauthorizedresponse.ProxyUnauthorizedResponse] = dataclasses.field(default=None)
    save_results: Optional[list[shared_saveresult.SaveResult]] = dataclasses.field(default=None)
    

File: SDKs/zuora.com/2021-08-20/typescript/src/sdk/models/operations/actionpostcreate.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class ActionPosTcreateQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=rejectUnknownFields" })
  rejectUnknownFields?: boolean;
}


export class ActionPosTcreateHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Zuora-WSDL-Version" })
  xZuoraWSDLVersion?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Zuora-Entity-Ids" })
  zuoraEntityIds?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Zuora-Track-Id" })
  zuoraTrackId?: string;
}


export class ActionPosTcreateRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: ActionPosTcreateQueryParams;

  @SpeakeasyMetadata()
  headers: ActionPosTcreateHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.ProxyActioncreateRequest;
}


export class ActionPosTcreateResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  headers: Record<string, string[]>;

  @SpeakeasyMetadata()
  proxyUnauthorizedResponse?: shared.ProxyUnauthorizedResponse;

  @SpeakeasyMetadata({ elemType: shared.SaveResult })
  saveResults?: shared.SaveResult[];

  @SpeakeasyMetadata()
  statusCode: number;
}
